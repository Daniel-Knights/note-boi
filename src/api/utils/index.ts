export * from './decorators';
export * from './error';
export * from './http';
