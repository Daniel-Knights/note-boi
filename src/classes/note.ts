import type Delta from 'quill-delta';

export class Note {
  readonly uuid: string = crypto.randomUUID();
  timestamp = Date.now();
  content: NoteContent = {
    title: '',
    body: '',
    delta: {
      ops: [],
    },
  };
}

type NoteContent = {
  title: string;
  body: string;
  delta: Partial<Delta>;
};
