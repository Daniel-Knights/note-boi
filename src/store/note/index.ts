export * from './event';
export * from './operations';
export * from './state';
export * from './types';
export * from './utils';
