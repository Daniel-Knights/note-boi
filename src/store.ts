import { reactive } from 'vue';
import { invoke } from '@tauri-apps/api/tauri';
import { v4 as uuidv4 } from 'uuid';

import { isEmptyNote } from './utils';

export class Note {
  readonly id = uuidv4();
  body = {
    delta: '',
    text: '',
  };
  timestamp = Date.now();
}

interface State {
  notes: Note[];
  selectedNote: Note;
}

export const state = reactive<State>({
  notes: [],
  selectedNote: new Note(),
});

/** Sorts notes in descending order by timestamp. */
function sortNotes() {
  state.notes.sort((a, b) => b.timestamp - a.timestamp);
}

/** Finds note index within {@link state.notes}. */
function findNoteIndex(id: string) {
  return state.notes.findIndex((nt) => nt.id === id);
}

/** Finds note within {@link state.notes}. */
export function findNote(id: string): Note | undefined {
  return state.notes.find((nt) => nt.id === id);
}

/** Deletes {@link state.selectedNote} when both `title` and `body` are empty. */
function clearEmptyNote(): void {
  const isValidClear = state.notes.length > 1;

  const foundNote = findNote(state.selectedNote.id);
  if (!foundNote) return;

  if (isValidClear && isEmptyNote(foundNote)) {
    deleteNote(state.selectedNote.id);
  }
}

const selectEvent = new CustomEvent('note-select');

/**
 * Looks for note with given `id` in {@link state.notes},
 * and sets it to {@link state.selectedNote}.
 */
export function selectNote(id: string): void {
  if (state.selectedNote.id === id) return;

  clearEmptyNote();

  const foundNote = findNote(id);
  if (foundNote) state.selectedNote = { ...foundNote };

  document.dispatchEvent(selectEvent);
}

/** Fetches all notes and updates {@link state}. */
export async function getAllNotes(): Promise<void> {
  const fetchedNotes = await invoke<Note[]>('get_all_notes').catch(console.error);

  const hasNotes = fetchedNotes && fetchedNotes.length > 0;
  if (!hasNotes) return newNote();

  state.notes = fetchedNotes;

  sortNotes();

  if (isEmptyNote(fetchedNotes[0])) {
    state.notes[0].timestamp = Date.now();
  }

  state.selectedNote = { ...state.notes[0] };

  document.dispatchEvent(selectEvent);
}

/** Deletes note with the given `id`. */
export function deleteNote(id: string): void {
  state.notes.splice(findNoteIndex(id), 1);
  state.selectedNote = { ...state.notes[0] };

  sortNotes();

  invoke('delete_note', { id }).catch(console.error);
}

/** Creates an empty note. */
export function newNote(): void {
  const foundNote = findNote(state.selectedNote.id);

  // Only update timestamp if selected note is empty
  if (foundNote && isEmptyNote(foundNote)) {
    state.selectedNote.timestamp = Date.now();
    return;
  }

  const freshNote = new Note();

  state.notes.unshift(freshNote);
  state.selectedNote = { ...freshNote };

  invoke('new_note', { note: { ...freshNote } }).catch(console.error);

  document.dispatchEvent(selectEvent);
}

/** Edits note body on Quill `text-change`. */
export function editBody(delta: string, text: string): void {
  const foundNote = findNote(state.selectedNote.id);
  if (!foundNote || text === foundNote.body.text) return;

  const timestamp = Date.now();
  foundNote.timestamp = timestamp;
  state.selectedNote.timestamp = timestamp;

  foundNote.body = { delta, text };

  sortNotes();

  invoke('edit_note', { note: { ...foundNote } }).catch(console.error);
}
