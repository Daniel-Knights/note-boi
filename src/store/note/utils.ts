import { Note } from '../../classes';
import { isEmptyNote } from '../../utils';

import { deleteNote } from './operations';
import { noteState } from './state';

/** Sorts notes in descending order by timestamp. */
export const sortNotesFn = (a: Note, b: Note): number => b.timestamp - a.timestamp;

/** Sorts {@link noteState.notes} in descending order by timestamp. */
export const sortStateNotes = (): Note[] => noteState.notes.sort(sortNotesFn);

/** Finds note index within {@link noteState.notes}. */
export function findNoteIndex(uuid?: string): number {
  return noteState.notes.findIndex((nt) => nt.uuid === uuid);
}

/** Finds note within {@link noteState.notes}. */
export function findNote(uuid?: string): Note | undefined {
  return noteState.notes.find((nt) => nt.uuid === uuid);
}

/**
 * Returns true if note is either {@link noteState.selectedNote}
 * or within {@link noteState.extraSelectedNotes}.
 */
export function isSelectedNote(note: Note): boolean {
  return (
    note.uuid === noteState.selectedNote.uuid ||
    noteState.extraSelectedNotes.some((nt) => nt?.uuid === note.uuid)
  );
}

/** Deletes {@link noteState.selectedNote} when note is empty. */
export function clearEmptyNote(): void {
  const isValidClear = noteState.notes.length > 1;
  if (!isValidClear) return;

  const foundNote = findNote(noteState.selectedNote.uuid);
  if (!foundNote) return;

  if (isEmptyNote(foundNote)) {
    deleteNote(noteState.selectedNote.uuid);
  }
}
