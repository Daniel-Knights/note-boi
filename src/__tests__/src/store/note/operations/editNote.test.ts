import * as a from '../../../../../api';
import * as n from '../../../../../store/note';
import { clearMockApiResults, mockApi } from '../../../../mock';
import { waitForAutoSync, waitUntil } from '../../../../utils';
import { mockUnsyncedEventCb, setupMockNoteEventListeners } from '../setup';

beforeAll(() => {
  setupMockNoteEventListeners();
});

describe('editNote', () => {
  it('Edits notes', async () => {
    const { calls } = mockApi();
    const debounceSyncSpy = vi.spyOn(a, 'debounceSync');

    await n.getAllNotes();

    const currentSelectedNote = { ...n.noteState.selectedNote };
    const noteToEdit = { ...n.findNote(n.noteState.selectedNote.uuid) };

    vi.clearAllMocks();
    clearMockApiResults({ calls });

    await waitForAutoSync(
      () => n.editNote({ ops: [{ insert: 'Title\nBody' }] }, 'Title', 'Body'),
      calls
    );

    expect(debounceSyncSpy).toHaveBeenCalledOnce();
    expect(mockUnsyncedEventCb).toHaveBeenCalledOnce();
    expect(mockUnsyncedEventCb).toHaveBeenCalledWith({
      kind: 'edited',
      note: noteToEdit.uuid,
    });

    const editedNote = n.findNote(noteToEdit.uuid)!;

    assert.notDeepEqual(n.noteState.selectedNote, currentSelectedNote);
    assert.notDeepEqual(editedNote, noteToEdit);
    // See `editNote` for why selectedNote content should remain the same
    assert.deepEqual(n.noteState.selectedNote.content, currentSelectedNote.content);
    assert.notStrictEqual(
      n.noteState.selectedNote.timestamp,
      currentSelectedNote.timestamp
    );
    assert.notStrictEqual(editedNote.timestamp, noteToEdit.timestamp);
    assert.strictEqual(calls.size, 1);
    assert.isTrue(calls.invoke.has('edit_note'));
  });

  it('Catches error', async () => {
    const { calls, setErrorValue } = mockApi();
    const consoleErrorSpy = vi.spyOn(console, 'error');

    await n.getAllNotes();

    vi.clearAllMocks();
    clearMockApiResults({ calls });
    setErrorValue.invoke('edit_note');

    n.editNote({ ops: [{ insert: 'Title\nBody' }] }, 'Title', 'Body');

    await waitUntil(() => calls.tauriApi.has('plugin:dialog|message'));

    expect(consoleErrorSpy).toHaveBeenCalledTimes(2);
    expect(consoleErrorSpy).toHaveBeenCalledWith('Note invoke error:');
    expect(consoleErrorSpy).toHaveBeenCalledWith(new Error('Mock Tauri Invoke error'));

    assert.strictEqual(calls.size, 1);
    assert.isTrue(calls.tauriApi.has('plugin:dialog|message'));
    assert.deepEqual(calls.tauriApi[0]!.calledWith, {
      message:
        'Something went wrong. Please try again or open an issue in the GitHub repo.',
      kind: 'error',
      okLabel: undefined,
      title: undefined,
    });
  });
});
