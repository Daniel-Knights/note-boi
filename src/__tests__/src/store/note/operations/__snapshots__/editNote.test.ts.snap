// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`editNote > Catches error > KeyStore.getKey() 1`] = `undefined`;

exports[`editNote > Catches error > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`editNote > Catches error > Storage.get('THEME') 1`] = `null`;

exports[`editNote > Catches error > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`editNote > Catches error > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`editNote > Catches error > Storage.get('USERNAME') 1`] = `null`;

exports[`editNote > Catches error > allCalls 1`] = `
[
  [
    "invoke",
    {
      "calledWith": {},
      "name": "get_all_notes",
    },
  ],
  [
    "tauriApi",
    {
      "calledWith": {
        "kind": "error",
        "message": "Something went wrong. Please try again or open an issue in the GitHub repo.",
        "okLabel": undefined,
        "title": undefined,
      },
      "name": "plugin:dialog|message",
    },
  ],
]
`;

exports[`editNote > Catches error > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`editNote > Catches error > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`editNote > Catches error > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [
    {
      "content": {
        "body": "Body",
        "delta": {
          "ops": [
            {
              "insert": "Title
Body",
            },
          ],
        },
        "title": "Title",
      },
      "timestamp": 0,
      "uuid": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "b9754a63-843c-4f59-9aea-9a374aa1603d",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "6b24cde1-6118-4277-9791-8eab3c71cabb",
    },
  ],
  "selectedNote": {
    "content": {
      "body": "😬ö",
      "delta": {
        "ops": [
          {
            "insert": "Note with special characters

😬ö
  sss
	
Ketchup
Salt 
Sugar

¯\\_(ツ)_/¯

──────────────────────────────────
──────▄▀▀▄────────────────▄▀▀▄────
─────▐▒▒▒▒▌──────────────▌▒▒▒▒▌───
─────▌▒▒▒▒▐─────────────▐▒▒▒▒▒▐───
────▐▒▒▒▒▒▒▌─▄▄▄▀▀▀▀▄▄▄─▌▒▒▒▒▒▒▌──
───▄▌▒▒▒▒▒▒▒▀▒▒▒▒▒▒▒▒▒▒▀▒▒▒▒▒▒▐───
─▄▀▒▐▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌───
▐▒▒▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▐───
▌▒▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌──
▒▒▐▒▒▒▒▒▒▒▒▒▄▀▀▀▀▄▒▒▒▒▒▄▀▀▀▀▄▒▒▐──
▒▒▌▒▒▒▒▒▒▒▒▐▌─▄▄─▐▌▒▒▒▐▌─▄▄─▐▌▒▒▌─
▒▐▒▒▒▒▒▒▒▒▒▐▌▐█▄▌▐▌▒▒▒▐▌▐█▄▌▐▌▒▒▐─
▒▌▒▒▒▒▒▒▒▒▒▐▌─▀▀─▐▌▒▒▒▐▌─▀▀─▐▌▒▒▒▌
▒▌▒▒▒▒▒▒▒▒▒▒▀▄▄▄▄▀▒▒▒▒▒▀▄▄▄▄▀▒▒▒▒▐
▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▄▄▄▒▒▒▒▒▒▒▒▒▒▒▐
▒▌▒▒▒▒▒▒▒▒▒▒▒▒▀▒▀▒▒▒▀▒▒▒▀▒▀▒▒▒▒▒▒▐
▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▀▒▒▒▄▀▄▒▒▒▀▒▒▒▒▒▒▒▐
▒▐▒▒▒▒▒▒▒▒▒▒▀▄▒▒▒▄▀▒▒▒▀▄▒▒▒▄▀▒▒▒▒▐
▒▓▌▒▒▒▒▒▒▒▒▒▒▒▀▀▀▒▒▒▒▒▒▒▀▀▀▒▒▒▒▒▒▐
▒▓▓▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌
▒▒▓▐▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌─
▒▒▓▓▀▀▄▄▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▐──
▒▒▒▓▓▓▓▓▀▀▄▄▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▄▄▀▀▒▌─
▒▒▒▒▒▓▓▓▓▓▓▓▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▒▒▒▒▒▐─
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌
▒▒▒▒▒▒▒█▒█▒█▀▒█▀█▒█▒▒▒█▒█▒█▒▒▒▒▒▒▐
▒▒▒▒▒▒▒█▀█▒█▀▒█▄█▒▀█▒█▀▒▀▀█▒▒▒▒▒▒▐
▒▒▒▒▒▒▒▀▒▀▒▀▀▒▀▒▀▒▒▒▀▒▒▒▀▀▀▒▒▒▒▒▒▐
█▀▄▒█▀▄▒█▀▒█▀█▒▀█▀▒█▒█▒█▒█▄▒█▒▄▀▀▐
█▀▄▒█▀▄▒█▀▒█▄█▒▒█▒▒█▀█▒█▒█▀██▒█▒█▐
▀▀▒▒▀▒▀▒▀▀▒▀▒▀▒▒▀▒▒▀▒▀▒▀▒▀▒▒▀▒▒▀▀▐
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▐

",
          },
        ],
      },
      "title": "Note with special characters",
    },
    "timestamp": 0,
    "uuid": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
  },
}
`;

exports[`editNote > Catches error > openedPopup 1`] = `undefined`;

exports[`editNote > Catches error > selectedTheme 1`] = `"System"`;

exports[`editNote > Catches error > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {
      "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
    },
    "new": "",
  },
  "username": "",
}
`;

exports[`editNote > Catches error > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`editNote > Edits notes > KeyStore.getKey() 1`] = `undefined`;

exports[`editNote > Edits notes > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`editNote > Edits notes > Storage.get('THEME') 1`] = `null`;

exports[`editNote > Edits notes > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`editNote > Edits notes > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`editNote > Edits notes > Storage.get('USERNAME') 1`] = `null`;

exports[`editNote > Edits notes > allCalls 1`] = `
[
  [
    "invoke",
    {
      "calledWith": {},
      "name": "get_all_notes",
    },
  ],
  [
    "invoke",
    {
      "calledWith": {
        "note": {
          "content": {
            "body": "Body",
            "delta": {
              "ops": [
                {
                  "insert": "Title
Body",
                },
              ],
            },
            "title": "Title",
          },
          "timestamp": 0,
          "uuid": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
        },
      },
      "name": "edit_note",
    },
  ],
]
`;

exports[`editNote > Edits notes > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`editNote > Edits notes > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`editNote > Edits notes > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [
    {
      "content": {
        "body": "Body",
        "delta": {
          "ops": [
            {
              "insert": "Title
Body",
            },
          ],
        },
        "title": "Title",
      },
      "timestamp": 0,
      "uuid": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "6fb3ce26-82e0-408f-84bf-bd8cd64c35b2",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "b9754a63-843c-4f59-9aea-9a374aa1603d",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "ac4e006d-38d4-4373-af31-4a5f46fc5184",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "2cbe0086-4b6f-466f-9ec0-9a302480da6a",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "2e36d871-0de2-43e6-9cbc-3c77e5d154cb",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "b8ff4ac5-6541-49d2-9c79-4444d872481e",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "9a2f52b3-69ac-4158-a9f7-b79ba9dbafd3",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "81648517-86ff-4ed9-8bf7-941c32dd0fad",
    },
    {
      "content": {
        "body": "amet consectetur adipisicing",
        "delta": {
          "ops": [
            {
              "insert": "Amet consectetur adipisicing\\n\\namet consectetur adipisicing\\namet consectetur adipisicing\\namet consectetur adipisicing\\n",
            },
          ],
        },
        "title": "Amet consectetur adipisicing",
      },
      "timestamp": 0,
      "uuid": "6b24cde1-6118-4277-9791-8eab3c71cabb",
    },
  ],
  "selectedNote": {
    "content": {
      "body": "😬ö",
      "delta": {
        "ops": [
          {
            "insert": "Note with special characters

😬ö
  sss
	
Ketchup
Salt 
Sugar

¯\\_(ツ)_/¯

──────────────────────────────────
──────▄▀▀▄────────────────▄▀▀▄────
─────▐▒▒▒▒▌──────────────▌▒▒▒▒▌───
─────▌▒▒▒▒▐─────────────▐▒▒▒▒▒▐───
────▐▒▒▒▒▒▒▌─▄▄▄▀▀▀▀▄▄▄─▌▒▒▒▒▒▒▌──
───▄▌▒▒▒▒▒▒▒▀▒▒▒▒▒▒▒▒▒▒▀▒▒▒▒▒▒▐───
─▄▀▒▐▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌───
▐▒▒▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▐───
▌▒▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌──
▒▒▐▒▒▒▒▒▒▒▒▒▄▀▀▀▀▄▒▒▒▒▒▄▀▀▀▀▄▒▒▐──
▒▒▌▒▒▒▒▒▒▒▒▐▌─▄▄─▐▌▒▒▒▐▌─▄▄─▐▌▒▒▌─
▒▐▒▒▒▒▒▒▒▒▒▐▌▐█▄▌▐▌▒▒▒▐▌▐█▄▌▐▌▒▒▐─
▒▌▒▒▒▒▒▒▒▒▒▐▌─▀▀─▐▌▒▒▒▐▌─▀▀─▐▌▒▒▒▌
▒▌▒▒▒▒▒▒▒▒▒▒▀▄▄▄▄▀▒▒▒▒▒▀▄▄▄▄▀▒▒▒▒▐
▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▄▄▄▒▒▒▒▒▒▒▒▒▒▒▐
▒▌▒▒▒▒▒▒▒▒▒▒▒▒▀▒▀▒▒▒▀▒▒▒▀▒▀▒▒▒▒▒▒▐
▒▌▒▒▒▒▒▒▒▒▒▒▒▒▒▀▒▒▒▄▀▄▒▒▒▀▒▒▒▒▒▒▒▐
▒▐▒▒▒▒▒▒▒▒▒▒▀▄▒▒▒▄▀▒▒▒▀▄▒▒▒▄▀▒▒▒▒▐
▒▓▌▒▒▒▒▒▒▒▒▒▒▒▀▀▀▒▒▒▒▒▒▒▀▀▀▒▒▒▒▒▒▐
▒▓▓▌▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌
▒▒▓▐▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌─
▒▒▓▓▀▀▄▄▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▐──
▒▒▒▓▓▓▓▓▀▀▄▄▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▄▄▀▀▒▌─
▒▒▒▒▒▓▓▓▓▓▓▓▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▒▒▒▒▒▐─
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▌
▒▒▒▒▒▒▒█▒█▒█▀▒█▀█▒█▒▒▒█▒█▒█▒▒▒▒▒▒▐
▒▒▒▒▒▒▒█▀█▒█▀▒█▄█▒▀█▒█▀▒▀▀█▒▒▒▒▒▒▐
▒▒▒▒▒▒▒▀▒▀▒▀▀▒▀▒▀▒▒▒▀▒▒▒▀▀▀▒▒▒▒▒▒▐
█▀▄▒█▀▄▒█▀▒█▀█▒▀█▀▒█▒█▒█▒█▄▒█▒▄▀▀▐
█▀▄▒█▀▄▒█▀▒█▄█▒▒█▒▒█▀█▒█▒█▀██▒█▒█▐
▀▀▒▒▀▒▀▒▀▀▒▀▒▀▒▒▀▒▒▀▒▀▒▀▒▀▒▒▀▒▒▀▀▐
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▐

",
          },
        ],
      },
      "title": "Note with special characters",
    },
    "timestamp": 0,
    "uuid": "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
  },
}
`;

exports[`editNote > Edits notes > openedPopup 1`] = `undefined`;

exports[`editNote > Edits notes > selectedTheme 1`] = `"System"`;

exports[`editNote > Edits notes > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {
      "7a7bccda-3fac-4465-b9c9-8f67dc1b5bdf",
    },
    "new": "",
  },
  "username": "",
}
`;

exports[`editNote > Edits notes > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;
